functions/go/create-setters/README.md

```diff
@@ -32,85 +32,122 @@ data:
   setter_name2: setter_value2
 ```
 
-If this function adds setter comments to fields for which you didn't intend 
-to parameterize, you can simply review and delete/modify those comments manually.
+On invoking `create-setters`, it performs the following steps:
+1. Segregates the input setters into scalar-setters and array-setters.
+2. Searches for the resource field values to be parameterized.
+3. Checks if there is any match considering the following cases.,
+   - For a scalar node, performs substring match with scalar setters.
+   - For an array node, checks if all values match with any of the array setters.
+4. Adds comments to the fields matching the setter values using setter names as parameters.
+
+>? If this function adds setter comments to fields for which you didn't intend to parameterize,
+you can simply review and delete/modify those comments manually.
 
 <!--mdtogo-->
 
 ### Examples
 
 <!--mdtogo:Examples-->
 
-#### Setting comment for scalar values
+### Setting comments for scalar nodes
 
 Let's start with the input resource in a package
 
 ```yaml
+# resources.yaml
 apiVersion: v1
 kind: Deployment
 metadata:
-  name: ubuntu-deployment 
+  name: ubuntu-deployment-1
```

Comment:
@phanimarupaka @mengqiy I am thinking it's better to use fields other than name/namespace (like `replicas`, `image tag` in this case) to show case this function. We should customize name/namespace using `ensure-name-substring`/`set-namespace` because those are more idiomatic ways to customize name/namespace. 

This might send wrong guidance to the end-users.